src/PKSim.Core/Snapshots/Mappers/DataInfoMapper.cs

```diff
@@ -38,9 +41,28 @@ private Task<SnapshotExtendedProperties> mapExtendedProperties(ModelExtendedProp
          return _extendedPropertiesMapper.MapToSnapshot(extendedProperties);
       }
 
-      public override Task<ModelDataInfo> MapToModel(SnapshotDataInfo snapshot)
+      public override async Task<ModelDataInfo> MapToModel(SnapshotDataInfo snapshot)
       {
-         throw new NotImplementedException();
+         var dataInfo = new ModelDataInfo(EnumHelper.ParseValue<ColumnOrigins>(snapshot.Origin))
+         {
+            AuxiliaryType = EnumHelper.ParseValue<AuxiliaryType>(snapshot.AuxiliaryType),
+            Category = snapshot.Category,
+            ComparisonThreshold = snapshot.ComparisonThreshold,
+            Date = snapshot.Date,
+            LLOQ = snapshot.LLOQ,
+            MolWeight = snapshot.MolWeight,
+            Source = snapshot.Source,
+         };
+
+         if(snapshot.ExtendedProperties != null)
+            dataInfo.ExtendedProperties.AddRange(await extendedPropertiesFrom(snapshot));
+
+         return dataInfo;
+      }
+
+      private async Task<IEnumerable<IExtendedProperty>> extendedPropertiesFrom(SnapshotDataInfo snapshot)
+      {
+         return await _extendedPropertiesMapper.MapToModel(snapshot.ExtendedProperties);
```

Comment:
if you do not use the return value, async and await adds complexity for nothing here.
Simply returns the task and remove async and await alltogether